### ISSUE TITLE
Pluggable template engine support for helm 3?

### ISSUE THREAD

**Issue by vroad** [helm/helm#9302]
https://github.com/helm/helm/issues/8290#issuecomment-642395892

Is this still true that helm provides a way for adding templating engines?

I'm not familiar with internal structure of helm, but looks like current implementation has no way to replace only go template renderer with something else.

https://github.com/helm/helm/blob/46d80f68c11d8107b688b9c8299fa02bccba312f/pkg/engine/engine.go#L250

helm 3 already already has a way to install plugins. Can we extend its functionality so that users can develop their own renderer implementation?

That would prevent other tools from using helm as a library though.
AFAIK terraform helm provider is one of such example. The provider doesn't need helm CLI to be installed because it incorporates helm as a library.

Developing many renderers in-tree might make it difficult to update dependencies and make CLI binary bloated.

Or... could plugin renderers run on server side? Maybe too much complicated to do so.

**Comment by bacongobbler** [comment]
> Is this still true that helm provides a way for adding templating engines?

This functionality was removed in Helm 3. Helm 2 was under active development for nearly 5 years, and no other template engine was ever introduced. Several community members advocated for new template engines. We pointed them towards [the `engine` mechanism](https://github.com/helm/helm/issues/2577#issuecomment-341548308), but none was ever written.

> helm 3 already already has a way to install plugins. Can we extend its functionality so that users can develop their own renderer implementation?

Plugins currently cannot inject themselves into another command like `helm install` or `helm upgrade`. They act more like git plugins in which they're standalone "helper" commands. `helm env` and `helm diff` are a few examples.
